### ISSUE TITLE
Merge solargraph-arc code into solargraph-rails

### ISSUE THREAD

**Issue by iftheshoefritz** [iftheshoefritz/solargraph-rails#28]
@alisnic will be joining solargraph-rails, and together we'll merge his excellent [solargraph-arc](https://github.com/alisnic/solargraph-arc) project with solargraph-rails. I'm sure users will be especially excited for improvements like db/schema.rb parsing that this will bring.

Large parts of the solargraph-rails code will be replaced in the process, but the interface to solargraph (and hence the experience for users) will remain the same.

**Comment by julianrubisch** [comment]
That is some great news! Congrats!

**Comment by alisnic** [comment]
I added note to solargraph-arc README and archived the repo 🚀

**Comment by emmohac** [comment]
Very excited to try this out.

You should also setup PayPal or something for donation because I couldn't find it.

**Comment by iftheshoefritz** [comment]
Comment in the Solargraph project's rails issue [announcing 1.0.0.pre.1 is here](https://github.com/castwide/solargraph/issues/87#issuecomment-1072533206)

**Comment by alisnic** [comment]
@iftheshoefritz I think it's time to publish a relase. Thoughts?

**Comment by iftheshoefritz** [comment]
@alisnic agreed. Two thoughts:

- builds are failing, let's fix those (but maybe that does't need to happen before release)
- let's update the README to match the solargraph-arc functionality

**Comment by iftheshoefritz** [comment]
I'm away next week, but can do a release in the second week of October.

**Comment by alisnic** [comment]
> Two thoughts

Makes sense, will do a PR to update README when I have some free time on my hands and will take a look at your PR. Regarding build errors though - last time I tried to fix it, I couldn't :( Something is off with Github CI env

**Comment by aelkoussy** [comment]
Hey, thanks for your great work! :) 
So now the plugin depends on schema.rb not the annotate comments? 
does that mean it won't work for people with structure.sql?

I tried the plugin but unfortunately I didn't get any completions (despite having the annotate comments)

Thanks for your efforts though!

**Comment by iftheshoefritz** [comment]
@aelkoussy  right now v1.0.0.pre.1 uses schema.rb, but in theory still has support for annotate. There is not yet any support for structure.sql.

Previous versions only use annotate, which is what you get if you do not specify the pre release version.

Note that we will shortly move to 1.0.0, at which point you'll get the same support as 1.0.0.pre.1 by default.

If you're having a problem with annotate support (or any other issue), would you mind raising a new issue with more details?

**Comment by aelkoussy** [comment]
@iftheshoefritz Thanks for explaining, and sure, I will raise an issue for that with more details, thanks

**Comment by iftheshoefritz** [comment]
@alisnic after fixing the README I released 1.0.1 (I did a bad release and yanked 1.0.0 🤦‍♂️) with everything from `main`. We can work on the builds at our leisure without worrying about the release.

**Comment by alisnic** [comment]
Awesome!
